content/contributing/development/ui/scss.rst

```diff
@@ -0,0 +1,259 @@
+
+.. _contributing/scss:
+
+===============
+SCSS Guidelines
+===============
+
+
+
+Syntax and Formatting
+=====================
+
+.. tabs::
+
+    .. code-tab:: html SCSS
+
+        .o_foo, .o_foo_bar,
+        .o_baz {
+            height: $o-statusbar-height;
+
+            .o_qux {
+                height: $o-statusbar-height * 0.5;
+            }
+        }
+
+        .o_corge {
+            background: $o-list-footer-bg-color;
+        }
+
+    .. code-tab:: css CSS
+
+        .o_foo, .o_foo_bar, .o_baz, .o_qux {
+            height: 32px;
+        }
+
+        .o_foo .o_quux, .o_foo_bar .o_quux, .o_baz .o_quux {
+            height: 16px;
+        }
+
+        .o_corge {
+            background: #EAEAEA;
+        }
+
+* four (4) space indents, no tabs;
+* 80 character wide columns;
+* related selectors on the **same** line; unrelated selectors on new lines;
+* empty space between the last selector and the opening brace (`{`).
+* the closing brace (`}`) on its own new line.
+* one line for each declaration;
+* meaningful use of whitespace.
+
+
+.. _scss/properties_order:
+
+Properties order
+================
+
+Order properties from the "outside", starting from `position` and ending with decorative rules (font,  filters...).
+
+:ref:`Scoped SCSS variables <scss/scoped-scss-var>` and `CSS variables <css_variables>`_ must be placed at the very top, followed by an empty line separating them from other declarations.
+
+
+.. code-block::
+
+    .o_myComponent {
+        $-inner-gap: $border-width + $legend-margin-bottom;
+
+        --myComponent-margin: 1rem;
+        --myComponent-size: 3rem;
+
+        @include o-position-absolute(1rem);
+        display: block;
+        margin: var(--myComponent-margin);
+        width: calc(var(--myComponent-size) + #{$-inner-gap});
+        border: 0;
+        padding: 1rem;
+        background: blue;
+        font-size: 1rem;
+        filter: blur(2px);
+    }
+
+Naming Conventions
+==================
+
+Naming conventions in CSS are hugely useful in making your code more strict, transparent and informative.
+
+
+Use the "Grandchild" approach
+-----------------------------
+Avoid creating hyper-specific classes & variables names.
+When naming nested elements, opt for the "Grandchild" approach.
```

Comment:
- Going to the line doesn't appear as such in the rendering, but only as a space. It's best to write it accordingly in the rst. Either stay on the same line, or make it a new paragraph
- Don't use `&` instead of "and" in a text. It's not grammatically correct.